sdk/dapp-kit/tsconfig.json

```diff
@@ -0,0 +1,16 @@
+{
+	"extends": "../build-scripts/tsconfig.shared.json",
+	"include": ["src"],
+	"compilerOptions": {
+		"module": "CommonJS",
+		"outDir": "dist/cjs",
+		"isolatedModules": true,
+		"jsx": "react-jsx",
+		"baseUrl": "..",
+		"paths": {
+			"@mysten/sui.js/transactions": ["../typescript/src/builder/export.ts"],
```

Comment:
n00b question: would project references make sense here?

Reply:
it's a little complicated.  project refs are nice because they actually build the project being referenced.  The issue is you need to run the build to get types to update, so you either need a dev server or a manual build if you want to keep things in sync if you are working on both projects.  To complicate things more, we want to build both esm and cjs, and with project refs you only get to build one typescript "project".  We are also using esbuild to build the js side of things, so project references would only generate the types. 

The js monorepo dev story in pretty complex these days, and I haven't found a set of options that provide an ideal experience, this is most in line with what we are doing for other projects right now

Reply:
thanks for the explainer! much appreciated 😄